pkg/sdk/sdk_test.go

```diff
@@ -0,0 +1,390 @@
+					err := r.ParseMultipartForm(10000) //File size to come from config
+					if err != nil {
+						response.ToJson(w, http.StatusBadRequest, codes.GetErr(codes.ErrFileSizeExceeded), nil)
+						log.Error(err.Error())
+						return
+					}
+					file, _, err := r.FormFile("file")
+					if err != nil {
+						response.ToJson(w, http.StatusBadRequest, codes.GetErr(codes.ErrFileParseFail), nil)
+						log.Error(err.Error())
+						return
+					}
+					defer file.Close()
+
+					response.ToJson(w, http.StatusCreated, "SUCCESS", map[string]interface{}{
+						"id": "1",
+					})
+				})
+				return svr
+			},
+			ValidateFunc: func(id string, err error) {
+				if err != nil {
+					t.Errorf("Want: %v, Got: %v", nil, err.Error())
+				}
+				if id != "1" {
+					t.Errorf("Want: %v, Got: %v", "not nil", "")
+				}
+			},
+			cleanupFunc: func(svr *httptest.Server) {
+				svr.Close()
+			},
+		},
+		{
+			name:     "Failure:: Register :: incorrect status code received",
+			filePath: "./../../docs/Failure.html",
+			setupFunc: func() *httptest.Server {
+				svr := testServer("/v1/register", http.MethodPost, func(w http.ResponseWriter, r *http.Request) {
+					response.ToJson(w, http.StatusBadRequest, "Failure", nil)
+				})
+				return svr
+			},
+			ValidateFunc: func(id string, err error) {
+				if err.Error() != "non success status code received : 400" {
+					t.Errorf("Want: %v, Got: %v", "non success status code received : 400", err.Error())
+				}
+			},
+			cleanupFunc: func(svr *httptest.Server) {
+				svr.Close()
+			},
+		},
+		{
+			name:     "Failure:: Register :: incorrect test server path", //doubt
+			filePath: "./../../go.mod",
+			setupFunc: func() *httptest.Server {
+				svr := testServer("new", http.MethodPost, func(w http.ResponseWriter, r *http.Request) {
+					response.ToJson(w, http.StatusCreated, "SUCCESS", nil)
+				})
+				return svr
+			},
+			ValidateFunc: func(id string, err error) {
+				if err.Error() != fmt.Errorf("non success status code received : %v", http.StatusNotFound).Error() {
+					t.Errorf("Want: %v, Got: %v", fmt.Errorf("non success status code received : %v", http.StatusNotFound), err)
+				}
+			},
+			cleanupFunc: func(svr *httptest.Server) {
+				svr.Close()
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			svr := tt.setupFunc()
+			defer tt.cleanupFunc(svr)
+
+			calls := NewHtmlToPdfSvc(svr.URL)
+			fileBytes, _ := os.ReadFile(tt.filePath)
+			id, err := calls.Register(fileBytes)
+
+			tt.ValidateFunc(id, err)
+		})
+	}
+}
+func Test_Replace(t *testing.T) {
+	tests := []struct {
+		name              string
+		filePath          string
+		id                string
+		setupFunc         func() *httptest.Server
+		mockServerHandler func(w http.ResponseWriter, r *http.Request)
+		ValidateFunc      func(err error)
+		cleanupFunc       func(*httptest.Server)
+		expectedResponse  model.Response
+	}{
+		{
+			name:     "Success:: Replace",
```

Comment:
Success case must validate that the file bytes sent from the sdk function is the same as the one received by the server along with headers sent in the request from the sdk function.

Reply:
done excepet headers are not matching

Reply:
done